streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -193,7 +194,7 @@ public void print(Serde<K> keySerde, Serde<V> valSerde) {
     public void print(Serde<K> keySerde, Serde<V> valSerde, String streamName) {
         String name = topology.newName(PRINTING_NAME);
         streamName = (streamName == null) ? this.name : streamName;
-        topology.addProcessor(name, new KeyValuePrinter<>(keySerde, valSerde, streamName), this.name);
+        topology.addProcessor(name, new KStreamPeek<>(new PrintForeachAction(null, keySerde, valSerde, streamName), false), this.name);
```

Comment:
If I use PrintWriter(System.out), I cannot know whether is System.out or not. Because if I call `PrintWriter(System.out)#close()`, the output will be muted by me :(